src/main/java/de/tum/in/www1/artemis/domain/lecture/ExerciseUnit.java

```diff
@@ -62,4 +59,25 @@ public ZonedDateTime getReleaseDate() {
     public void setReleaseDate(ZonedDateTime releaseDate) {
         // Should be set in associated exercise
     }
+
+    @Override
+    public Set<LearningGoal> getLearningGoals() {
+        return exercise == null || !Hibernate.isPropertyInitialized(exercise, "learningGoals") ? new HashSet<>() : exercise.getLearningGoals();
```

Comment:
Is there a known case where the property `learningGoals``is not initialized or is this just as precaution?

Reply:
Yes, “learningGoals” in the exercise is a lazy collection. So there might be cases where they are not initialized.

Reply:
Does it make sense to load the `learningGoals` if this happens - or is it fine that we just pretend there are none?

Reply:
I did not realize that this method is within the domain object so we can not load it here - If this turns out to be a problem in the future, we could move this logic to a service/repository.